.ado/setVersionEnvVars.js

```diff
@@ -3,72 +3,48 @@ const fs = require('fs');
 const path = require('path');
 const child_process = require('child_process');
 const semver = require('semver');
-const pkgJsonPath = path.resolve(__dirname, "../vnext/package.json");
-
-// Helper to format npmVersion in a way that the Version.rc resource files want it
-function npmVersionToRcVersion(npmVersion) {
-  let groups = npmVersion.split(/[\.-]/);
-  const [major,minor,patch,_junk,prerelease] = groups;
-  return `${major},${minor},${patch},${prerelease ? prerelease : '0'}`;
-}
 
-let pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, "utf8"));
+const pkgJsonPath = path.resolve(__dirname, "../vnext/package.json");
+const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, "utf8"));
 
 // Record commit number, so that additional build tasks can record that commit in the NuGet
 const commitId = child_process.execSync(`git rev-list HEAD -n 1`).toString().trim();
 
-let versionEnvVars = {
+const versionEnvVars = {
   RNW_PKG_VERSION_STR: pkgJson.version,
   RNW_PKG_VERSION_MAJOR: semver.major(pkgJson.version),
   RNW_PKG_VERSION_MINOR: semver.minor(pkgJson.version),
   RNW_PKG_VERSION_PATCH: semver.patch(pkgJson.version),
   npmVersion: pkgJson.version,
-  publishCommitId: commitId
+  publishCommitId: commitId,
+  reactNativeDevDependency: pkgJson.devDependencies['react-native'],
 }
 
 // Set the build number so the build in the publish pipeline and the release pipeline are named with the convenient version
 // See: https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#updatebuildnumber-override-the-automatically-generated-build-number
```

Comment:
Should consider splitting this out and not running it as part of PRs

Reply:
These will overlap, but we still get the PR name and the previous build number wasn't super useful either, so I think this is probably fine?